packages/gatsby/src/utils/parcel/__tests__/compile-gatsby-files.ts

```diff
@@ -20,6 +20,29 @@ const dir = {
   gatsbyNodeAsDirectory: `${__dirname}/fixtures/gatsby-node-as-directory`,
 }
 
+jest.mock(`gatsby-worker`, () => {
+  const gatsbyWorker = jest.requireActual(`gatsby-worker`)
+
+  const { WorkerPool: OriginalWorkerPool } = gatsbyWorker
+
+  class WorkerPoolThatCanUseTS extends OriginalWorkerPool {
+    constructor(workerPath: string, options: any) {
+      options.env = {
+        ...(options.env ?? {}),
+        NODE_OPTIONS: `--require ${require.resolve(
+          `../../worker/__tests__/test-helpers/ts-register.js`
+        )}`,
+      }
+      super(workerPath, options)
+    }
+  }
+
+  return {
+    ...gatsbyWorker,
+    WorkerPool: WorkerPoolThatCanUseTS,
+  }
+})
+
```

Comment:
this is just a trick to be able to "load" TS file that is used in tests - in actually built package, regular CJS will be loaded